packages/flutter/test/material/raised_button_test.dart

```diff
@@ -617,6 +617,39 @@ void main() {
     expect(box.size, equals(const Size(76, 36)));
     expect(childRect, equals(const Rect.fromLTRB(372.0, 293.0, 428.0, 307.0)));
   });
+
+  testWidgets('RaisedButton.icon responds to applied padding', (WidgetTester tester) async {
+    const Key buttonKey = Key('test');
+    const Key labelKey = Key('label');
+    await tester.pumpWidget(
+      Directionality(
+        textDirection: TextDirection.ltr,
+        child: Material(
+          child: Center(
+            child: RaisedButton.icon(
+              icon: const Icon(Icons.add),
+              padding: const EdgeInsets.all(16),
+              key: buttonKey,
+              onPressed: () {},
+              label: const Text(
+                'Hello',
+                key: labelKey,
+              ),
+            ),
+          ),
+        ),
+      ),
+    );
+
+    final Size buttonSize = tester.getSize(find.byKey(buttonKey));
+    final Size labelSize = tester.getSize(find.byKey(labelKey));
+
+    // By default the padding between icon and label is 8.0, and default
+    // size of Icon is 24 pixels * 24 pixels, hence
+    // height = 24 + 16 * 2 = 56 and
+    // width = 24 + width of Text + 8 + 16 * 2
+    expect(buttonSize, Size(labelSize.width + 8 + 24 + 32, 56));
```

Comment:
After thinking about this a little, I think it might be better to use Rects and compare the distances between the edges of the padding and the edges of the label and icon. WDYT? Something like (also, this is just pseudocode, I'm not absolutely sure of the syntax):
```dart
// Icon is the left-most widget in a [RaisedButton] when textDirection is set to [TextDirection.ltr].
expect(tester.getRect(find.byType(Padding)).left, tester.getRect(find.byIcon(Icons.add)).left + 16.0);
// same thing for every edge.
```

Reply:
Sounds good, I guess we can check the both, the size of the button after padding is applied and also check the positions of the children.

Reply:
If you want to use the Size, we should then be a little more accurate here with where the numbers are derived from.
```dart
    expect(
      buttonSize,
      Size(
        // Button width = Padding between icon and label + icon width + label width + Left padding + Right padding
        (labelRect.left - iconRect.right) + iconRect.width + labelRect.width + 16.0 + 16.0
        // Icon height is used here because the icon's height is taller than the label height.
        // Button height = Icon height + Top padding + Bottom padding
        iconRect.height + 16.0 + 16.0,
      ),
    );
```

However, It does feel like all we're doing here is testing the exact same thing as below, so I'm unsure if having both is particularly meaningful.

Reply:
Makes sense. We will only check the Icon and Label position.